modal/io_streams.py

```diff
@@ -140,12 +141,14 @@ def __init__(
             self._consume_container_process_task = asyncio.create_task(self._consume_container_process_stream())
 
     @property
-    def file_descriptor(self):
+    def file_descriptor(self) -> int:
+        """Possible values are `1` for stdout and `2` for stderr."""
         return self._file_descriptor
 
     async def read(self) -> T:
-        """Fetch and return contents of the entire stream. If EOF was received,
-        return an empty string.
+        """Fetch and return contents of the entire stream.
+
+        If EOF was received, return an empty string.
```

Comment:
This was here before, but this text is a bot confusing. Technically EOF is always received, in which case it returns the full buffer. Maybe something about it waits for the stream to close would be more appropriate.